src/charm.py

```diff
@@ -124,15 +124,18 @@ def _on_get_certificate_action(self, event: ActionEvent) -> None:
             return
 
         tls_rel_data = tls_certificates_relation.data[self.app]
-        if not tls_rel_data.get(f"certificate-{hostname}"):
-            event.fail("Certificate not available")
+        if any(
+            not tls_rel_data.get(key)
+            for key in [f"certificate-{hostname}", f"ca-{hostname}", f"chain-{hostname}"]
+        ):
+            event.fail("Missing or incomplete certificate data")
             return
 
         event.set_results(
             {
-                f"certificate-{hostname}": tls_rel_data[f"certificate-{hostname}"],
-                f"ca-{hostname}": tls_rel_data[f"ca-{hostname}"],
-                f"chain-{hostname}": tls_rel_data[f"chain-{hostname}"],
+                f"certificate-{hostname}": tls_rel_data.get(f"certificate-{hostname}"),
+                f"ca-{hostname}": tls_rel_data.get(f"ca-{hostname}"),
+                f"chain-{hostname}": tls_rel_data.get(f"chain-{hostname}"),
```

Comment:
I think the above check for data availability makes the use of `.get` here unnecessary.

Reply:
Yes, but I think we should still use `.get()` instead of `[]` since it's objectively safer since it's unconditional and never raises an Exception

Reply:
Gotcha, but what is the downside of an exception here? If we expect the keys to exist (which you do with the above check), I would rather have an exception thrown if a key is missing than have invalid relation data which is much harder to debug.

Reply:
Gotcha, but what is the downside of an exception here? If we expect the keys to exist (which you do with the above check), I would rather have an exception thrown if a key is missing than return invalid data, as this alerts us to invalid programming logic.

Reply:
I wouldn't say a down side, but rather added complexity in that we'd need to add exception handling to  the potential `KeyError` that might be thrown, which will seem like added code for not much value. And we'd probably have to add unit test for coverage as well.

Reply:
I discussed about this internally and I'll revert back to using `[]`, the idea is indeed to let the exception flow and alert us of that really strange situations. I'll do it in a follow-up PR.